apollo-common/src/main/java/com/ctrip/framework/apollo/common/controller/GlobalDefaultExceptionHandler.java

```diff
@@ -153,4 +156,44 @@ private void printLog(String message, Throwable ex, Level logLevel) {
     Tracer.logError(ex);
   }
 
+  /**
+   * 获取请求参数
+   * @param request
+   * @return
+   */
+  private Map getInputParams(HttpServletRequest request) {
+    Map<String, String> map = new HashMap();
+    Enumeration paramNames = request.getParameterNames();
+    while (paramNames.hasMoreElements()) {
+      String paramName = (String) paramNames.nextElement();
+      String[] paramValues = request.getParameterValues(paramName);
+      if (paramValues.length == 1) {
+        String paramValue = paramValues[0];
+        if (paramValue.length() != 0) {
+          map.put(paramName, paramValue);
+        }
+      }
+    }
+    return map;
+  }
+
+  /**
+   * 获取 ip 地址
+   * @param request
+   * @return
+   */
+  public String getIpAddr(HttpServletRequest request) {
+    String ip = request.getHeader("x-forwarded-for");
+    if (ip == null || ip.length() == 0 || "unknown".equalsIgnoreCase(ip)) {
```

Comment:
why would ip equals to "unknown?"

Reply:
If the service is represented by some agents such as **Squid**，and The **forwarded_for** attribute is set to **off**，then **X-Forwarded-For** equals “unknown”

Reference link：https://www.cnblogs.com/softidea/p/5705297.html

Reply:
Thanks for the knowledge sharing, actually apollo already has similar logic in `com.ctrip.framework.apollo.configservice.controller.ConfigController#tryToGetClientIp` which only considers the standard case, and I think maybe it's enough.  Because it's hard to enumerate all the proxy headers in the world...